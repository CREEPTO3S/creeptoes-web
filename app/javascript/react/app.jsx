import React, { useRef } from 'react';
import { BrowserRouter as Router, Route, Switch } from 'react-router-dom';
import { CleanWrapper } from '@layouts';
import { FourOFour } from '@pages';

const App = () => {
  const constraintsRef = useRef(null);

  return (
    <CleanWrapper innerRef={constraintsRef}>
      <Router>
        <Switch>
          <Route path="*">
            <FourOFour constraintsRef={constraintsRef} />
          </Route>
        </Switch>
      </Router>
    </CleanWrapper>
  );
};

export default App;
