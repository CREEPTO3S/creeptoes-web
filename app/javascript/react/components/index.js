import CommandPrompt from '@components/command-prompt';

export {
  // eslint-disable-next-line import/prefer-default-export
  CommandPrompt,
};
