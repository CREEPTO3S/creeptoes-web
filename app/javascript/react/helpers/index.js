export const WINDOW_TYPE_ENUMS = {
  warning: 'WARNING',
  error: 'ERROR',
};

export const AVAILABLE_COMMANDS = {
  help: {
    cmd: 'help',
    desc: 'list available commands',
  },
  clear: {
    cmd: 'clear',
    desc: 'clear screen',
  },
};
