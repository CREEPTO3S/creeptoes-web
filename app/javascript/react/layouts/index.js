import CleanWrapper from '@layouts/clean-wrapper';
import Window from '@layouts/window';

export {
  CleanWrapper,
  Window,
};
