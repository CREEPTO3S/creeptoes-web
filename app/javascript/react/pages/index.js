import FourOFour from '@pages/404';
import Home from '@pages/home';

export {
  // eslint-disable-next-line import/prefer-default-export
  FourOFour,
  Home,
};
