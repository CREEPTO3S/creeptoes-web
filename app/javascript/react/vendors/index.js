import DarkMode from '@vendors/dark-mode';

export {
  // eslint-disable-next-line import/prefer-default-export
  DarkMode,
};
