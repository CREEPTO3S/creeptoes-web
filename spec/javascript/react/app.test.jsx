import React from 'react';
import { render } from '@testing-library/react';

import App from '@root/app';

describe('#render', () => {
  it('renders properly', () => {
    const { container } = render(<App />);

    expect(container).toMatchSnapshot();
  });
});
