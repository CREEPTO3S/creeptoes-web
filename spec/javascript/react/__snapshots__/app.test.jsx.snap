// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`#render renders properly 1`] = `
<div>
  <div
    class="sc-bdvvaa cWwyBq"
  >
    <div
      draggable="false"
      style="user-select: none;"
    >
      <div
        class="sc-gsDJrp fObuKD window"
      >
        <div
          class="sc-hKwCoD title-bar"
        >
          <div
            class="sc-eCImvq title-bar-text"
          >
            Error
          </div>
          <div
            class="sc-jRQAMF title-bar-controls"
          >
            <button
              aria-label="Minimize"
              class="sc-gKckTs"
              type="button"
            />
            <button
              aria-label="Maximize"
              class="sc-gKckTs"
              type="button"
            />
            <button
              aria-label="Close"
              class="sc-gKckTs"
              type="button"
            />
          </div>
        </div>
        <div
          class="sc-dkPtyc window-body"
        >
          <p
            class="sc-iCfLBT bCgVbS"
          >
            404 Not Found.
          </p>
        </div>
      </div>
    </div>
  </div>
</div>
`;
